keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+         *
+         * @return A SecretClient with the options set from the builder.
+         * @throws IllegalStateException If {@link Builder#credentials(ServiceClientCredentials)}
+         * @throws IllegalStateException If {@link Builder#credentials(ServiceClientCredentials)}
+         * has not been set.
+         */
+        public SecretClient build() {
+            if (credentials == null) {
+                throw new IllegalStateException("'credentials' is required.");
+            }
+
+            if (vaultEndPoint == null){
+                throw new IllegalStateException("'Vault's Endpoint Url' is required.");
+            }
+
+            // Closest to API goes first, closest to wire goes last.
+            final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+            policies.add(new UserAgentPolicy(userAgent));
+            policies.add(new RequestIdPolicy());
+            policies.add(retryPolicy);
+            policies.add(new KeyvaultCredentialsPolicy(credentials, vaultEndPoint));
+
+            policies.addAll(this.policies);
+
+            policies.add(new HttpLoggingPolicy(httpLogDetailLevel));
+
+            HttpPipeline pipeline = httpClient == null
+                    ? new HttpPipeline(policies)
+                    : new HttpPipeline(httpClient, policies);
+
+            return new SecretClient(vaultEndPoint, pipeline);
+        }
+
+        /**
+         * Sets the credentials to use when authenticating HTTP requests.
+         *
+         * @param vaultEndPoint The vault endpoint url is used as destination on Azure to send requests to.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code credentials} is {@code null}.
+         */
+        public Builder vaultEndPoint(String vaultEndPoint) {
+            Objects.requireNonNull(vaultEndPoint);
+            this.vaultEndPoint = vaultEndPoint;
+            return this;
+        }
+
+        /**
+         * Sets the credentials to use when authenticating HTTP requests.
+         *
+         * @param credentials The credentials to use for authenticating HTTP requests.
+         * @return The updated Builder object.
+         * @throws NullPointerException if {@code credentials} is {@code null}.
+         */
+        public Builder credentials(ServiceClientCredentials credentials) {
+            Objects.requireNonNull(credentials);
+            this.credentials = credentials;
+            return this;
+        }
+
+        /**
+         * Sets the logging level for HTTP requests and responses.
+         *
+         * @param logLevel The amount of logging output when sending and receiving HTTP requests/responses.
+         * @return The updated Builder object.
+         */
+        public Builder httpLogDetailLevel(HttpLogDetailLevel logLevel) {
+            httpLogDetailLevel = logLevel;
+            return this;
+        }
```

Comment:
No null check, no throws statement

Reply:
This is optional. 
Default value of HttpLogDetailLevel.NONE is set.
Added that in javadoc.
Added Null check too.